### PR TITLE
Shift grass zonal stats script to standalone

### PR DESCRIPTION
**Pull request by hblitza**
This PR refactores the grass zonal script:
- enables standalone running of the script
- enables multi geometries

### CODE REVIEW

File: pygeoapi/process_scripts/standalone_grass_zonal_stats.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
+import os
+import subprocess
+import sys
+import tempfile
+import csv
+import json
+from argparse import ArgumentParser
+
+gisbase = subprocess.check_output(["grass", "--config", "path"], text=True).strip()
```
Comment (JakobMiksch) [inline pygeoapi/process_scripts/standalone_grass_zonal_stats.py:9]: would be great to add some comments what is happening here

File: pygeoapi/process_scripts/standalone_grass_zonal_stats.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+# Import GRASS Python bindings
+import grass.script.setup as gsetup
+from grass.pygrass.modules.shortcuts import general as g
+from grass.pygrass.modules.shortcuts import raster as r
+from grass.pygrass.modules.shortcuts import vector as v
+
+
+def generate_zonal_stats(rastermap, geometries):
+    with tempfile.TemporaryDirectory() as tempdir:
```
Comment (JakobMiksch) [inline pygeoapi/process_scripts/standalone_grass_zonal_stats.py:25]: please add docs for the function

File: pygeoapi/process_scripts/standalone_grass_zonal_stats.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+                g.remove(type='vector', name=polygon_name, flags='f')
+                g.remove(type='raster', name=zone_name, flags='f')
+
+            session.finish()
+        with open(f'{tempdir}/output', 'r') as result:
+            result = result.readlines()
+
+        # format result to JSON
+        # TODO: only works for one polygon and is a quick fix that should changed
```
Comment (JakobMiksch) [inline pygeoapi/process_scripts/standalone_grass_zonal_stats.py:69]: the second part of the sentence is not clear to me

**Review by JakobMiksch — COMMENTED** [review]
Looks good so far. I did not run it yet. See some code comments inside.

File: pygeoapi/processes/zonal_statistics_grass.py [hunk 4]
```diff
@@ -88,76 +83,14 @@ def __init__(self, processor_def):
         super().__init__(processor_def, PROCESS_METADATA)
 
     def execute(self, data):
+        rasterURL = data.get('rasterURL', None)
```
Comment (JakobMiksch) [inline pygeoapi/processes/zonal_statistics_grass.py:86]: variable should be in snake_case like `raster_url`

**Review by JakobMiksch — COMMENTED** [review]
Looks good to me. I tried it and it works. 
One question to clarify inline, otherwise we can merge it.

File: pygeoapi/processes/algorithms/geometries_test.json [hunk 5]
```diff
@@ -0,0 +1,28 @@
+[
```
[the remark this answers is not in the document]
Comment (hblitza) [reply pygeoapi/processes/algorithms/geometries_test.json:1]: yes. i will move it or rename it

File: pygeoapi/processes/zonal_statistics_grass.py [hunk 6]
```diff
@@ -1,15 +1,8 @@
...
-from grass.pygrass.modules.shortcuts import vector as v
-from grass.pygrass.modules import Module
-import grass.script.setup as gsetup
+
 from pygeoapi.process.base import BaseProcessor
-import tempfile
-import csv
-import json
+
+from pygeoapi.processes.algorythms.standalone_grass_zonal_stats import generate_zonal_stats
```
[the remark this answers is not in the document]
Comment (hblitza) [reply pygeoapi/processes/zonal_statistics_grass.py:5]: good question, will check

**Pull request merged**